C code doesn't compile with gcc 4.8

```
cc -Wall -Werror -I/usr/local/include -o objecthash_test objecthash_test.c objecthash.c crypto-algorithms/sha256.c -L/usr/local/lib -ljson-c
In file included from objecthash_test.c:1:0:
objecthash.h:10:14: error: variably modified ‘hash’ at file scope
 typedef byte hash[HASH_SIZE];
              ^
objecthash_test.c: In function ‘a_to_hash’:
objecthash_test.c:8:3: error: ‘for’ loop initial declarations are only allowed in C99 mode
   for (int n = 0; n < HASH_SIZE; ++n)
   ^
objecthash_test.c:8:3: note: use option -std=c99 or -std=gnu99 to compile your code
In file included from objecthash.c:6:0:
objecthash.h:10:14: error: variably modified ‘hash’ at file scope
 typedef byte hash[HASH_SIZE];
              ^
In file included from /usr/include/json-c/linkhash.h:16:0,
                 from /usr/include/json-c/json.h:22,
                 from objecthash.c:2:
objecthash.c: In function ‘object_hash_dict’:
objecthash.c:56:40: error: variable ‘val’ set but not used [-Werror=unused-but-set-variable]
     json_object_object_foreach(d, key, val) {
                                        ^
objecthash.c:56:35: error: variable ‘key’ set but not used [-Werror=unused-but-set-variable]
     json_object_object_foreach(d, key, val) {
                                   ^
objecthash.c: In function ‘object_hash_list’:
objecthash.c:148:3: error: ‘for’ loop initial declarations are only allowed in C99 mode
   for (int n = 0; n < len; ++n) {
   ^
objecthash.c:148:3: note: use option -std=c99 or -std=gnu99 to compile your code
cc1: all warnings being treated as errors
make: *** [objecthash_test] Error 1
```

Adding -std=c99 should fix this. Does leave another minor problem, tho:

objecthash.c: In function 'object_hash_dict':
objecthash.c:60:40: error: variable 'val' set but not used [-Werror=unused-but-set-variable]
     json_object_object_foreach(d, key, val) {

This still seems to be broken even though the command includes `-std=c99`.

Here's what I get when I run `make c`:

```
cc -fPIC -shared -std=c99 -Wall -Werror -o libobjecthash.so objecthash.c -lcrypto `pkg-config --libs --cflags icu-uc json-c`
In file included from objecthash.c:14:0:
objecthash.h:13:14: error: variably modified ‘hash’ at file scope
 typedef byte hash[HASH_SIZE];
              ^
In file included from /usr/include/json-c/linkhash.h:16:0,
                 from /usr/include/json-c/json.h:22,
                 from objecthash.c:2:
objecthash.c: In function ‘object_hash_dict’:
objecthash.c:62:40: error: variable ‘val’ set but not used [-Werror=unused-but-set-variable]
     json_object_object_foreach(d, key, val) {
                                        ^
objecthash.c:62:35: error: variable ‘key’ set but not used [-Werror=unused-but-set-variable]
     json_object_object_foreach(d, key, val) {
                                   ^
objecthash.c:66:3: error: implicit declaration of function ‘alloca’ [-Werror=implicit-function-declaration]
   byte *hashes = alloca(2 * len * sizeof(hash));
   ^
objecthash.c:66:18: error: initialization makes pointer from integer without a cast [-Werror]
   byte *hashes = alloca(2 * len * sizeof(hash));
                  ^
objecthash.c: In function ‘unicode_normalize_string’:
objecthash.c:259:17: error: initialization makes pointer from integer without a cast [-Werror]
     UChar *us = alloca(length * sizeof(UChar));
                 ^
objecthash.c:265:18: error: initialization makes pointer from integer without a cast [-Werror]
     UChar *usn = alloca(uslength * 10 * sizeof(UChar));
                  ^
objecthash.c:275:16: error: initialization makes pointer from integer without a cast [-Werror]
     char *n8 = alloca(n8length);
                ^
cc1: all warnings being treated as errors
make: *** [libobjecthash.so] Error 1
```

--- a/objecthash.c
+++ b/objecthash.c
@@ -61,9 +61,12 @@ static bool object_hash_dict(/*const*/ json_object * const d, hash h) {
   {
     json_object_object_foreach(d, key, val) {
       ++len;
+      // key and val are unused, but we don't want the compiler to complain.
+      (void)key;
+      (void)val;
     }
   }
-  byte *hashes = alloca(2 * len * sizeof(hash));
+  byte *hashes = malloc(2 * len * sizeof(hash));
 
   size_t n = 0;
   json_object_object_foreach(d, key, val) {
@@ -74,6 +77,9 @@ static bool object_hash_dict(/*const*/ json_object * const d, hash h) {
 
   qsort(hashes, len, 2 * sizeof(hash), dict_comp);
   hash_bytes('d', hashes, 2 * len * sizeof(hash), h);
+
+  free(hashes);
+
   return true;
 }
 
@@ -256,13 +262,13 @@ static json_object *unicode_normalize_string(json_object *j) {
     u_strFromUTF8(NULL, 0, &length, s, strlen(s), &err);
     assert(!err);
 
-    UChar *us = alloca(length * sizeof(UChar));
+    UChar *us = malloc(length * sizeof(UChar));
     int32_t uslength;
     u_strFromUTF8(us, length, &uslength, s, strlen(s), &err);
     assert(!err);
 
     // FIXME: surely we can find out how long the result will be?
-    UChar *usn = alloca(uslength * 10 * sizeof(UChar));
+    UChar *usn = malloc(uslength * 10 * sizeof(UChar));
     int32_t usnlength = unorm2_normalize(un, us, uslength, usn, 10 * uslength,
                                        &err);
     assert(!err);
@@ -272,9 +278,13 @@ static json_object *unicode_normalize_string(json_object *j) {
     u_strToUTF8(NULL, 0, &n8length, usn, usnlength, &err);
     assert(!err);
 
-    char *n8 = alloca(n8length);
+    char *n8 = malloc(n8length);
     u_strToUTF8(n8, n8length, &n8length, usn, usnlength, &err);
     assert(!err);
+
+    free(us);
+    free(usn);
+    free(n8);
 
     return json_object_new_string(n8);
   }

--- a/objecthash.h
+++ b/objecthash.h
@@ -1,14 +1,13 @@
-#include <stdbool.h>
 #include <openssl/sha.h>
+#include <stdbool.h>
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
-
 typedef unsigned char byte;
 
-static const int HASH_SIZE = SHA256_DIGEST_LENGTH;
+#define HASH_SIZE SHA256_DIGEST_LENGTH
 
 typedef byte hash[HASH_SIZE];
 typedef SHA256_CTX hash_ctx;

